docs/core-concepts/versioning.md

```diff
@@ -1,6 +1,201 @@
+
+Versioning Scheme
+=================
+
+.NET Core can be defined as and will version in the following way:
+
+- A runtime and framework implementation, distributed as packages. Each package is versioned independently.
+- A set of metapackage packages that reference and make the fine-grained packages easier to reference and use together as a versioned unit. Each metapackage has its own versioning scheme.
+- A set of target frameworks (for example, netstandard) that represent a progressively larger API set, described in a set of versioned snapshots.
+
+Packages
+--------
+
+Library packages evolve and version independently. Packages that overlap with .NET Framework System.\* assemblies typically use 4.x versions, aligning with the .NET Framework 4.x versioning (a historical choice). Packages that do not overlap with the .NET Framework libraries (for example, System.Metadata) typically start at 1.0 and increment from there.
+
+The lower-level packages (most of the ones with 4.x versions) are treated specially due to being at the base of the platform. Explanation:
+
+- The lower-level packages will typically version as a set, since they have implementation-level dependencies between them. 
+- APIs will typically only be added to the lower-level packages as part of a minor or major .NET Core release since adding APIs would require adding a new netstandard version, which currently requires an update to the NuGet client. As a result, these packages will only be updated with patch versions (no new APIs) between .NET Core releases.
+
+Metapackages
+-------------
+
+Versioning for .NET Core metapackages is based on the target framework that they map to. The metapackages adopt the version number of the highest version (for example, netstandard1.5) of the target framework it maps to in its package closure. The patch version for the metapackage is used to represent updates to the metapackage to reference updated packages (that don't include a new target framework version). As a result, the metapackages are not strictly SemVer compliant because their versioning scheme doesn't represent the degree of change in the underlying packages, but only API level. 
+
+There are two primary metapackages for .NET Core.
+
+**NETStandard.Library**
+
+- v1.5 as of .NET Core 1.0.
+- Maps to `netstandard` target framework. 
+- Describes the packages that are considered required for modern app development and that .NET platforms must implement to be considered a ".NET Standard" platform.
+
+**Microsoft.NETCore.App**
+
+- v1.0 as of .NET Core 1.0 (versions intended to match).
+- Maps to `netcoreapp` target framework.
+- Describes the packages in the .NET Core distribution.
+
+Note: It is reasonable to reference the `Microsoft.NETCore.App` metapackage but use the netstandard framework. You will get access to all of the netstandard-compatible assets from those packages, which will be a subset of all the packages but more than are referenced via `NETStandard.Library`.
```

Comment:
I don't think all users will have a clear understanding of which one is bigger when they get to this note. Maybe you can call it out earlier or start the note by saying which metapackage contains just a subset of the libraries. 
I'm thinking something like: 
Note: Since the `NETStandard.Library` contains contains fewer packages, it is reasonable to reference the `Microsoft.NETCore.App` metapackage but use the netstandard framework. By doing that, you'll get access to all of the netstandard-compatible assets from those packages, which will be a subset of all the packages included in the `Microsoft.NETCore.App` metapackage but more than are referenced via `NETStandard.Library`.

I'm also curious to understand why this happens. :smile: